scripts/pi-hole/js/queries.js

```diff
@@ -78,7 +78,7 @@ $(function () {
       var replyid = parseInt(data[5], 10);
       // DNSSEC status
       var dnssecStatus;
-      var ede = data[11];
+      var ede = data[11] ? data[11] : "";
```

Comment:
I just wonder is this the "proper" way to do this JavaScript? Test if an object at a possibly noon-existing offset in the array exist? Shouldn't it rather be a `data.length > 10` kind of test?

I'm only asking because this test could lead to undefined behavior (or even a crash) in other programming languages but I'm by no means a JavaScript expert same this may be less of an issue with interpreted script languages.

Reply:
You mean `> 11` then 😉. However, not sure if `data.length` may be `12` when the array key was somehow defined, but with a null value? `data[11]` explicitly checks the value, which is required for `data[11].length` to not throw an error.

What I did test is that checking for an array value like this does not throw an error when the key was never defined. Only the array itself need to be defined, but that is true for `data.length` as well.